src/v/raft/consensus.h

```diff
@@ -338,6 +339,19 @@ class consensus {
     voter_priority next_target_priority();
     voter_priority get_node_priority(model::node_id id) const;
 
+    /**
+     * Return true if there is no state backing this consensus group i.e. there
+     * is no snapshot and log is empty
+     */
+    bool is_initial_state() const {
+        static constexpr model::offset not_initialized{};
```

Comment:
i'd remove the `static` here and make it `const`, or use `model::offset{}` inline in the three lines below. @Lazin pointed out the other day that the compiler may insert additional code to deal with static scope initialization here.

Reply:
this is constexpr and the class is a value type, so it propagates throughout. it's just a naked int. 

but yeah, doesn't matter if static. i'd +1 @Lazin's idea.

Reply:
ah, right. somewhere along the way i got the incorrect notion stuck in my that constexpr didn't imply const